File: src/za/co/adhd_developers/java/questions/Question48.java
package za.co.adhd_developers.java.questions;

import za.co.adhd_developers.java.tools.Question;

import java.math.BigInteger;

public class Question48 implements Question {

    private BigInteger answer = BigInteger.ZERO;

    @Override
    public void doWork() {
        BigInteger runningTotal = BigInteger.ZERO;
        for (int n = 1; n <= 1000; n++) {
            runningTotal = runningTotal.add(BigInteger.valueOf(n).pow(n));
        }

        this.answer = runningTotal.mod(BigInteger.valueOf(10000000000L));
    }

    @Override
    public void printAnswer() {
        System.out.println("=================================");
        System.out.println("Question 48");
        System.out.println("The last 10 digits of the series, 1^1 + 2^2 + 3^3 + ... + 1000^1000, is: " + this.answer.toString());
        System.out.println("=================================");
    }
}

File: src/za/co/adhd_developers/kotlin/questions/Question48.kt
package za.co.adhd_developers.kotlin.questions

import za.co.adhd_developers.kotlin.Question
import java.math.BigInteger

class Question48 : Question {

    private var answer = BigInteger.ZERO

    override fun doWork() {
        var finalTotal = BigInteger.ZERO
        (1L..1000L).forEach {
            finalTotal = finalTotal.plus(BigInteger.valueOf(it).pow(it.toInt()))
        }

        this.answer = finalTotal.mod(BigInteger.valueOf(10000000000L))
    }

    override fun printAnswer() {
        println("=================================")
        println("Question 48")
        println("The last 10 digits of the series, 1^1 + 2^2 + 3^3 + ... + 1000^1000, is: " + this.answer.toString())
        println("=================================")
    }
}